### DISCUSSION TITLE
How to change the figsize of  the light curve fitting figure ( by forward model )

### DISCUSSION THREAD

**Discussion by d11122** [phoebe-project/phoebe2#485] [Questions]
Hello,I want to change the figsize of he light curve fitting figure ( by forward model ), it is produced by these code in phoebe : 

_ = b.plot(kind='lc', model='emcee_posts', x='phases', y='fluxes', 
           s={'dataset': 0.005},
           marker={'dataset': '.'})
_ = b.plot(kind='lc', model='emcee_posts', x='phases', y='residuals', 
           z={'dataset': 0, 'model': 1},
           save='figure_posteriors_sample_from_lc.pdf', show=True)

But I do not like the figsize with default, I checked the api : 'b.plot' of phoebe, however i get nothing. Then I try set the plt.rc like this:
plt.rc('figure',figsize=[6.0, 4.0])
but it is not useful.

what method can chage the figsize?
thank you!   --daimin

**Comment by kecnry** [comment]
Hi Daimin - 

You can pass a matplotlib figure object to [b.plot](http://phoebe-project.org/docs/latest/api/phoebe.parameters.ParameterSet.plot) (technically you only need to pass it for the call with `show` or `save`).

For example:

```
import phoebe
import matplotlib.pyplot as plt
b = phoebe.default_binary()
b.add_dataset('lc', compute_times=phoebe.linspace(0,1,101))
b.run_compute()

mplfig = plt.figure(figsize=(10,3))
b.plot(fig=mplfig, show=True)
```

Note that this is currently buggy when used with `animate=True` and is something that will hopefully improve in the future.

-Kyle

**Comment by d11122** [comment]
thank you! Kyle, I will do it with your method.


&nbsp;





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "phoebe-project/phoebe2"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年7月3日(星期六) 晚上9:25
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [phoebe-project/phoebe2] How to change the figsize of  the light curve fitting figure ( by forward model ) (#485)





 
Hi Daimin -
 
You can pass a matplotlib figure object to b.plot (technically you only need to pass it for the call with show or save).
 
For example:
 import phoebe import matplotlib.pyplot as plt b = phoebe.default_binary() b.add_dataset('lc', compute_times=phoebe.linspace(0,1,101)) b.run_compute() mplfig = plt.figure(figsize=(10,3)) b.plot(fig=mplfig, show=True)  
Note that this is currently buggy when used with animate=True and is something that will hopefully improve in the future.
 
-Kyle
